osquery/sql/sqlite_filesystem.cpp

```diff
@@ -0,0 +1,166 @@
+/**
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <boost/filesystem.hpp>
+#include <boost/optional.hpp>
+
+#include <sqlite3.h>
+
+namespace osquery {
+  
+static boost::optional<std::string> findExistingProgramPathFromCommand(
+    const char *path, char escape_symbol, bool allow_quoting, bool shortest) {
+  size_t length = strlen(path);
+  std::string result;
+  size_t pos = 0;
+  // Skip spaces
+  for (; pos < length; ++pos) {
+    if (!isspace(path[pos])) {
```

Comment:
It seems to me like we could possibly leverage [some boost trim](http://thispointer.com/how-to-trim-strings-in-c-using-boost-string-algorithm-library/) here to make this a little cleaner and more robust. You're just hoping to find the first non-space occurance with this logic yes?